src/test/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessorIT.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Predicate;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+import org.opensearch.client.Response;
+import org.opensearch.neuralsearch.utils.TestHelper;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+
+public class TextEmbeddingProcessorIT extends OpenSearchRestTestCase {
+
+    private static final String indexName = "text_embedding_index";
+
+    private static final String pipelineName = "pipeline-hybrid";
+
+    private final ClassLoader classLoader = this.getClass().getClassLoader();
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public void test_text_embedding_processor() throws IOException {
+        String modelId = uploadModel();
+        loadModel(modelId);
+        createPipelineProcessor(modelId);
+        createIndex();
+        ingestDocument();
+    }
+
+    private String uploadModel() throws IOException {
+        Response uploadResponse = TestHelper.makeRequest(
+            client(),
+            "POST",
+            "/_plugins/_ml/models/_upload",
+            null,
+            TestHelper.toHttpEntity(
+                FileUtils.readFileToString(new File(classLoader.getResource("processor/UploadModelRequestBody.json").getFile()), "utf-8")
+            ),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, "Kibana"))
+        );
+        JsonNode uploadResJson = objectMapper.readTree(EntityUtils.toString(uploadResponse.getEntity()));
+        String taskId = uploadResJson.get("task_id").asText();
+        assertNotNull(taskId);
+
+        JsonNode taskQueryResult = getTaskQueryResponse(taskId);
+        boolean isComplete = checkComplete(taskQueryResult);
+        while (!isComplete) {
+            taskQueryResult = getTaskQueryResponse(taskId);
+            isComplete = checkComplete(taskQueryResult);
+        }
+        String modelId = taskQueryResult.get("model_id").asText();
+        assertNotNull(modelId);
+        return modelId;
+    }
+
+    private void loadModel(String modelId) throws IOException {
```

Comment:
This is a great common function which will be used by Query too, move this function to an abstract class named: BaseNeuralSearchIT.class and then extend the new class to use these functions.

The IT for Query can then use these common functions.